libraries/AP_InertialSensor/AP_InertialSensor_ZeroOnex.cpp

```diff
@@ -0,0 +1,760 @@
+    }
+    dev->transfer(wbuf, ptr, wbuf, ptr);
+
+    for(i=0;i<size;i++)
+    {
+        *(value+i) = wbuf[5+i];
+    }
+}
+
+void AP_InertialSensor_ZeroOnex::fpga_write_reg8(uint16_t reg, uint8_t value) 
+{
+    uint8_t wbuf[8];
+    uint8_t ptr = 0;
+
+    wbuf[ptr++] = FuncBit_Write | FuncBit_Reg | FuncBit_Bit8;
+    wbuf[ptr++] = reg & 0xff;
+    wbuf[ptr++] = (reg >> 8) & 0xff;
+    wbuf[ptr++] = value;
+
+    dev->transfer(wbuf, ptr, wbuf, ptr);
+}
+uint8_t AP_InertialSensor_ZeroOnex::fpga_read_reg8(uint16_t reg) 
+{
+    uint8_t wbuf[8];
+    uint8_t ptr = 0;
+
+    wbuf[ptr++] = FuncBit_Read | FuncBit_Reg | FuncBit_Bit8;
+    wbuf[ptr++] = reg & 0xff;
+    wbuf[ptr++] = (reg >> 8) & 0xff;
+    wbuf[ptr++] = 0xff;
+    wbuf[ptr++] = 0xff;
+    dev->transfer(wbuf, ptr, wbuf, ptr);
+    return wbuf[4];
+}
+
+void AP_InertialSensor_ZeroOnex::fpga_write_reg16(uint16_t reg, uint16_t value) 
+{
+    uint8_t wbuf[8];
+    uint8_t ptr = 0;
+
+    wbuf[ptr++] = FuncBit_Write | FuncBit_Reg | FuncBit_Bit16;
+    wbuf[ptr++] = reg & 0xff;
+    wbuf[ptr++] = (reg >> 8) & 0xff;
+    wbuf[ptr++] = value & 0xff;
+    wbuf[ptr++] = (value >> 8) & 0xff;
+    
+
+    dev->transfer(wbuf, ptr, wbuf, ptr);
+}
+
+void AP_InertialSensor_ZeroOnex::fpga_write_ram8(uint16_t reg, uint16_t ram_addr, uint8_t* value, uint16_t size) {
+    uint8_t wbuf[32];
+    uint8_t ptr = 0;
+    uint8_t i;
+    if (value ==nullptr) return;
+    wbuf[ptr++] = FuncBit_Write | FuncBit_Ram | FuncBit_Bit8;
+    wbuf[ptr++] = reg & 0xff;
+    wbuf[ptr++] = (reg >> 8) & 0xff;
+    wbuf[ptr++] = ram_addr & 0xff;
+    wbuf[ptr++] = (ram_addr >> 8) & 0xff;
+        for(i=0;i<size&&ptr<32;i++)
+    {
+        wbuf[ptr++] = *(value+i);
+    }
+        if(ptr == 32)
+    {
+        GCS_SEND_TEXT(MAV_SEVERITY_WARNING, "Buff ready to be overflow");
+    }
+    dev->transfer(wbuf, ptr, wbuf, ptr);
+}
+
+void AP_InertialSensor_ZeroOnex::fpga_read_ram8(uint16_t reg, uint16_t ram_addr, uint8_t* value, uint16_t size) {
+    uint8_t wbuf[32];
+    uint8_t ptr = 0;
+    uint8_t i;  
+    if (value ==nullptr) return;
+    wbuf[ptr++] = FuncBit_Read | FuncBit_Ram | FuncBit_Bit8;
+    wbuf[ptr++] = reg & 0xff;
+    wbuf[ptr++] = (reg >> 8) & 0xff;
+    wbuf[ptr++] = ram_addr & 0xff;
+    wbuf[ptr++] = (ram_addr >> 8) & 0xff;
+    wbuf[ptr++] = 0xff;
+        for(i=0;i<size&&ptr<32;i++)
+    {
+        wbuf[ptr++] = *(value+i);
+    }
+    
+    dev->transfer(wbuf, ptr, wbuf, ptr);
+
+    for(i=0;i<size;i++)
+    {
+        *(value+i) = wbuf[6+i];
+    }
+}
+
+uint8_t AP_InertialSensor_ZeroOnex::fpga_test(void) {
+    uint16_t wreg, rreg = 0;
+    wreg = rand() & 0xffff;
+    fpga_write_reg16(Addr_RW16_TestReg, wreg);
+    rreg = fpga_read_reg16(Addr_RW16_TestReg);
+    if (wreg != rreg) 
+    {
+        GCS_SEND_TEXT(MAV_SEVERITY_WARNING, "failed,wreg is:%d,rreg is:%d",wreg,rreg);
+        return 0;
+    }
+    GCS_SEND_TEXT(MAV_SEVERITY_INFO, "success,wreg is:%d,rreg is:%d",wreg,rreg);
```

Comment:
too many verbose status texts, please reduce for production use, or put behind an #if  (this is not the only one, check all GCS_SEND_TEXT)